### PR TITLE
SDIT-2215 POST identifying marks mappings

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/prisonperson/identifyingmarks/IdentifyingMarkImageResource.kt [hunk 1]
```diff
@@ -132,20 +123,5 @@ class IdentifyingMarkImageResource(private val service: IdentifyingMarkImageServ
...
-  ) = try {
-    service.createIdentifyingMarkImageMapping(mapping.toEntity())
-  } catch (e: DuplicateKeyException) {
-    throw DuplicateMappingException(
-      messageIn = "Identifying mark image mapping already exists",
-      duplicate = mapping,
-      existing = getExistingMappingSimilarTo(mapping),
-      cause = e,
-    )
-  }
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/prisonperson/identifyingmarks/IdentifyingMarkImageResource.kt:144]: Moved the duplicate handling to the service layer

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/prisonperson/identifyingmarks/IdentifyingMarkResource.kt [hunk 2]
```diff
@@ -87,5 +89,41 @@ class IdentifyingMarkResource(private val service: IdentifyingMarkService) {
   )
   suspend fun getIdentifyingMarkMappings(
     @PathVariable dpsId: UUID,
-  ) = service.getIdentifyingMarkMappings(dpsId).map { it.toDto() }
+  ) = service.getIdentifyingMarkMappings(dpsId)
+
+  @PostMapping("/identifying-mark")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/prisonperson/identifyingmarks/IdentifyingMarkResource.kt:94]: The main change for this PR is to create new endpoint to create identifying marks mappings

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/prisonperson/identifyingmarks/IdentifyingMarkImageService.kt [hunk 3]
```diff
@@ -1,19 +1,45 @@
...
     repository.findByDpsId(dpsImageId)
+      ?.toDto()
+      ?: throw NotFoundException("Identifying mark image mapping not found for DPS image id $dpsImageId")
 
-  suspend fun createIdentifyingMarkImageMapping(mapping: IdentifyingMarkImageMapping) {
-    repository.save(mapping)
+  suspend fun createIdentifyingMarkImageMapping(mapping: IdentifyingMarkImageMappingDto) {
+    try {
+      repository.save(mapping.toEntity())
+    } catch (e: DuplicateKeyException) {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/prisonperson/identifyingmarks/IdentifyingMarkImageService.kt:29]: Interesting, I thought that didn't work - something to do with commits and reactive - very odd

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**